core/unit_test/CMakeLists.txt

```diff
@@ -260,11 +260,63 @@ if(Kokkos_ENABLE_CUDA)
   )
 endif()
 
-KOKKOS_ADD_EXECUTABLE_AND_TEST(
-  UnitTest_Default
-  SOURCES
-    UnitTestMainInit.cpp
-    default/TestDefaultDeviceType.cpp
+if(Kokkos_ENABLE_HIP)
```

Comment:
Do you want to do it this way around instead of what I did with OpenMPTarget?

Reply:
No strong opinion.  I feel likes its good enough for now

Reply:
ok